### ISSUE TITLE
Propagate async Pub/Sub errors into Stream error channel

### ISSUE THREAD

**Issue by elefeint** [spring-cloud/spring-cloud-gcp#2501]
Per Spring Cloud Stream `errorChannelEnabled` [docs](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/current/reference/html/spring-cloud-stream.html#_producer_properties),

>  if the binder supports asynchroous send results, send failures are sent to an error channel for the destination.

We don't use the `errorChannel` parameter when creating producer message handler at all right now.

**Comment by dinvlad** [comment]
Has this been addressed? In other words, could one add `errorChannel` handler that will be invoked instead of nacking messages? Thanks

**Comment by elefeint** [comment]
This was done for the subscription side (for a Spring Cloud Stream sink), but not for publishing side (for a Spring Cloud Stream source).

Which one is your usecase?

**Comment by dinvlad** [comment]
Thanks @elefeint , I've tried this but with a global `errorChannel` instead and it isn't working for me, for some reason.

**Comment by elefeint** [comment]
@dinvlad Replacing [this line](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/fd7387c9f7770f325dce9fde8de78702022be25c/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-sample/src/main/java/com/example/SinkExample.java#L52) with `@ServiceActivator(inputChannel = "errorChannel")` captures the error for me. There are two things that could be tripping you up:
1) Spring Cloud Stream won't consider an error handed off to the global error handler "handled", and will propagate an error back to the binder.
2) If a stream-specific `my-topic.my-group.errors` has a listener, there is no longer an error to propagate to the global channel; the process completes successfully.

In the future, please create a new issue with any relevant information about what you tried and what the expected outcome was -- this makes it easier for developers to find useful threads in the future.

**Comment by dinvlad** [comment]
OK, thanks for clarifying @elefeint - I was under the impression I could get a global "catch all", but looks like that's not how it works. I'm assuming this behavior is not specific to PubSub?

**Comment by elefeint** [comment]
Right; this is how Spring Cloud Stream works -- here is a relevant [SO explanation](https://stackoverflow.com/a/56698724/10441462).
